src/main/native/compression/IntelInflater.cc

```diff
@@ -174,12 +173,47 @@ JNIEXPORT jint JNICALL Java_com_intel_gkl_compression_IntelInflater_inflateNativ
         env->ReleasePrimitiveArrayCritical(inputBuffer, next_in, 0);
         env->ReleasePrimitiveArrayCritical(outputBuffer, next_out, 0);
 
-        if (ret == ISAL_END_INPUT && lz_stream->avail_in == 0) {
+        if (ret == ISAL_DECOMP_OK && lz_stream->avail_in == 0) {
           env->SetBooleanField(obj, FID_inf_finished, true);
         }
 
-        return bytes_out;
+        if (ret != ISAL_DECOMP_OK) {
+          const char* msg;
+
+          switch (ret) {
+            case ISAL_INVALID_BLOCK:
+              msg = "Invalid deflate block found.";
+              break;
+            case ISAL_NEED_DICT:
+              msg = "Stream needs a dictionary to continue.";
+              break;
+            case ISAL_INVALID_SYMBOL:
+              msg = "Invalid deflate symbol found.";
+              break;
+            case ISAL_INVALID_LOOKBACK:
+              msg = "Invalid lookback distance found.";
+              break;
+            case ISAL_INVALID_WRAPPER:
+              msg = "Invalid gzip/zlib wrapper found.";
+              break;
+            case ISAL_UNSUPPORTED_METHOD:
+              msg = "Gzip/zlib wrapper specifies unsupported compress method.";
+              break;
+            case ISAL_INCORRECT_CHECKSUM:
+              msg = "Incorrect checksum found.";
+              break;
+            default:
+              msg = "isal_inflate returned an unknown return code.";
+              DBG("lsal_inflate returned %d", ret);
+          }
+
+          env->ExceptionClear();
+          env->ThrowNew(env->FindClass("java/lang/RuntimeException"), msg);
+          return -1;
+        }
 
+        int bytes_out = outputBufferLength - lz_stream->avail_out;
```

Comment:
Please declare bytes_out at the top this function.

Reply:
Doing this would introduce stylistic inconsistency with the rest of the codebase.

On a personal note, I dislike the ANSI C style of declaring variables at the top of each block because it makes it:
- more difficult to track the type of the variable,
- more difficult to track the first time the variable is used in a meaningful way,
- possible to use the variable by mistake before it is assigned a meaningful value.